File: frontend/src/data/seed.ts
export const USERS = [
  { id: 'u1', name: 'Sarah Chen', email: 'donor@test.com', password: 'donor123', role: 'donor' as const, trust_score: 92, avatar: 'SC' },
  { id: 'u2', name: 'HealthForAll Foundation', email: 'ngo@test.com', password: 'ngo123', role: 'ngo' as const, trust_score: 88, avatar: 'HF' },
  { id: 'u3', name: 'James Okafor', email: 'verifier@test.com', password: 'verifier123', role: 'verifier' as const, trust_score: 95, avatar: 'JO' },
];

export type User = typeof USERS[number];
export type Role = 'donor' | 'ngo' | 'verifier' | 'admin';

export const CAMPAIGNS = [
  {
    id: 'c1', ngo_id: 'u2', ngo_name: 'HealthForAll Foundation',
    title: 'Mobile Health Van — Rural Clinics',
    description: 'Deploying a fully equipped mobile health van to provide free medical checkups, vaccinations, and emergency care to 12 underserved rural villages across the northern corridor. Each visit includes basic diagnostics, maternal health screenings, and pediatric care.',
    category: 'Healthcare',
    image: 'https://images.pexels.com/photos/263402/pexels-photo-263402.jpeg?auto=compress&cs=tinysrgb&w=800',
    goal: 50000, total_raised: 34500, released: 18000, locked: 12500, refunded: 4000,
    status: 'active' as const, trust_score: 87, failure_count: 1, donors_count: 142, created_at: '2025-11-15',
  },
  {
    id: 'c2', ngo_id: 'u2', ngo_name: 'CleanWaterNow',
    title: 'Solar-Powered Water Purification',
    description: 'Installing solar-powered water purification systems in 8 communities lacking access to clean drinking water. Systems will serve approximately 4,000 people and run entirely on renewable energy.',
    category: 'Water & Sanitation',
    image: 'https://images.pexels.com/photos/2990644/pexels-photo-2990644.jpeg?auto=compress&cs=tinysrgb&w=800',
    goal: 75000, total_raised: 62000, released: 45000, locked: 17000, refunded: 0,
    status: 'active' as const, trust_score: 94, failure_count: 0, donors_count: 287, created_at: '2025-09-22',
  },
  {
    id: 'c3', ngo_id: 'u2', ngo_name: 'SafeHaven',
    title: "Women's Emergency Shelter Expansion",
    description: "Expanding capacity of the downtown emergency shelter from 40 to 80 beds, adding a childcare wing and job training center. Priority intake for domestic violence survivors and families with children.",
    category: 'Shelter & Safety',
    image: 'https://images.pexels.com/photos/8613089/pexels-photo-8613089.jpeg?auto=compress&cs=tinysrgb&w=800',
    goal: 120000, total_raised: 98000, released: 72000, locked: 26000, refunded: 0,
    status: 'active' as const, trust_score: 96, failure_count: 0, donors_count: 412, created_at: '2025-07-10',
  },
  {
    id: 'c4', ngo_id: 'u2', ngo_name: 'EduFuture',
    title: 'Youth STEM Lab — After-School Program',
    description: 'Building a community STEM lab with 3D printers, robotics kits, and coding stations serving 200 students aged 12-18. After-school programs run five days a week with certified instructors.',
    category: 'Education',
    image: 'https://images.pexels.com/photos/8613319/pexels-photo-8613319.jpeg?auto=compress&cs=tinysrgb&w=800',
    goal: 40000, total_raised: 28000, released: 15000, locked: 10000, refunded: 3000,
    status: 'active' as const, trust_score: 82, failure_count: 1, donors_count: 93, created_at: '2025-12-01',
  },
];

export type Campaign = typeof CAMPAIGNS[number];

export const MILESTONES = [
  { id: 'm1', campaign_id: 'c1', title: 'Vehicle Procurement & Equipping', description: 'Purchase and fully outfit the mobile health van with medical equipment.', amount: 18000, due_date: '2026-01-15', status: 'released' as const, evidence: 'Purchase receipts, vehicle registration, and equipment inventory photos uploaded.', reviewer_notes: 'All documentation verified and cross-checked with vendor invoices.' },
  { id: 'm2', campaign_id: 'c1', title: 'Phase 1 — Villages 1-3', description: 'Deploy to first 3 villages. Minimum 300 patients served.', amount: 8000, due_date: '2026-03-01', status: 'submitted' as const, evidence: 'Service logs, patient count records, and village chief sign-offs uploaded.', reviewer_notes: '' },
  { id: 'm3', campaign_id: 'c1', title: 'Phase 2 — Villages 4-6', description: 'Second deployment with maternal health services added.', amount: 8000, due_date: '2026-05-01', status: 'locked' as const, evidence: '', reviewer_notes: '' },
  { id: 'm4', campaign_id: 'c1', title: 'Phase 3 — Villages 7-9', description: 'Third deployment with dental screening.', amount: 8000, due_date: '2026-07-01', status: 'locked' as const, evidence: '', reviewer_notes: '' },
  { id: 'm5', campaign_id: 'c1', title: 'Phase 4 — Villages 10-12 & Final Report', description: 'Final deployment plus comprehensive impact report.', amount: 8000, due_date: '2026-09-01', status: 'locked' as const, evidence: '', reviewer_notes: '' },
  { id: 'm6', campaign_id: 'c2', title: 'Equipment Procurement', description: 'Purchase solar panels and purification units.', amount: 25000, due_date: '2025-11-01', status: 'released' as const, evidence: 'All equipment delivered and photos submitted.', reviewer_notes: 'Verified against purchase orders.' },
  { id: 'm7', campaign_id: 'c2', title: 'Installation Phase 1 (4 communities)', description: 'Install systems in first 4 communities.', amount: 20000, due_date: '2026-01-15', status: 'released' as const, evidence: 'Installation photos and community leader sign-offs.', reviewer_notes: 'Confirmed operational.' },
  { id: 'm8', campaign_id: 'c2', title: 'Installation Phase 2 (4 communities)', description: 'Complete remaining 4 installations.', amount: 17000, due_date: '2026-04-01', status: 'locked' as const, evidence: '', reviewer_notes: '' },
];

export type Milestone = typeof MILESTONES[number];
export type MilestoneStatus = 'locked' | 'submitted' | 'approved' | 'released' | 'rejected';

export const DONATIONS = [
  { id: 'd1', donor_id: 'u1', campaign_id: 'c1', campaign_title: 'Mobile Health Van — Rural Clinics', ngo_name: 'HealthForAll Foundation', amount: 500, released_portion: 260, refundable_portion: 240, date: '2026-01-20', status: 'active' as const },
  { id: 'd2', donor_id: 'u1', campaign_id: 'c2', campaign_title: 'Solar-Powered Water Purification', ngo_name: 'CleanWaterNow', amount: 1000, released_portion: 725, refundable_portion: 275, date: '2026-02-05', status: 'active' as const },
  { id: 'd3', donor_id: 'u1', campaign_id: 'c3', campaign_title: "Women's Emergency Shelter Expansion", ngo_name: 'SafeHaven', amount: 250, released_portion: 184, refundable_portion: 66, date: '2026-02-14', status: 'active' as const },
  { id: 'd4', donor_id: 'u1', campaign_id: 'c4', campaign_title: 'Youth STEM Lab — After-School Program', ngo_name: 'EduFuture', amount: 750, released_portion: 401, refundable_portion: 349, date: '2026-03-01', status: 'active' as const },
];

export type Donation = typeof DONATIONS[number];

export const NGOS = [
  { name: 'HealthForAll', initials: 'HF', tagline: 'Healthcare Access', funded: '$2.4M', campaigns: 12 },
  { name: 'CleanWaterNow', initials: 'CW', tagline: 'Clean Water Initiative', funded: '$1.8M', campaigns: 8 },
  { name: 'SafeHaven', initials: 'SH', tagline: 'Shelter & Safety', funded: '$3.1M', campaigns: 15 },
  { name: 'EduFuture', initials: 'EF', tagline: 'Education First', funded: '$2.7M', campaigns: 10 },
  { name: 'GreenEarth', initials: 'GE', tagline: 'Climate Action', funded: '$1.5M', campaigns: 6 },
  { name: 'FoodBank+', initials: 'FB', tagline: 'Food Security', funded: '$2.9M', campaigns: 14 },
  { name: 'TechForGood', initials: 'TG', tagline: 'Digital Literacy', funded: '$1.2M', campaigns: 5 },
  { name: 'YouthRise', initials: 'YR', tagline: 'Youth Empowerment', funded: '$1.9M', campaigns: 9 },
];

export const PLATFORM_STATS = {
  total_protected: '$12.4M',
  success_rate: '91%',
  total_campaigns: 44,
  total_donors: 3200,
  monthly_donations: [
    { month: 'Oct', amount: 145000 },
    { month: 'Nov', amount: 198000 },
    { month: 'Dec', amount: 267000 },
    { month: 'Jan', amount: 312000 },
    { month: 'Feb', amount: 289000 },
    { month: 'Mar', amount: 341000 },
  ],
  category_breakdown: [
    { name: 'Healthcare', value: 35 },
    { name: 'Education', value: 25 },
    { name: 'Water', value: 20 },
    { name: 'Shelter', value: 15 },
    { name: 'Other', value: 5 },
  ],
  escrow_status: [
    { name: 'Released', value: 5200000, color: '#FFFFFF' },
    { name: 'Locked', value: 4800000, color: '#FF6432' },
    { name: 'Refunded', value: 2400000, color: '#EF4444' },
  ],
};

export const VERIFICATION_QUEUE = [
  {
    id: 'vq1',
    milestone_id: 'm2',
    campaign_id: 'c1',
    campaign_title: 'Mobile Health Van — Rural Clinics',
    ngo_name: 'HealthForAll Foundation',
    milestone_title: 'Phase 1 — Villages 1-3',
    amount: 8000,
    submitted_at: '2026-03-15',
    evidence: 'Service logs, patient count records, and village chief sign-offs uploaded.',
    status: 'submitted' as const,
  },
];

export const ACTIVITY_FEED = [
  { id: 'a1', type: 'donation', message: 'Anonymous donated $500', time: '2 hours ago', campaign_id: 'c1' },
  { id: 'a2', type: 'milestone', message: 'Milestone "Vehicle Procurement" released', time: '3 days ago', campaign_id: 'c1' },
  { id: 'a3', type: 'donation', message: 'Sarah C. donated $250', time: '5 days ago', campaign_id: 'c1' },
  { id: 'a4', type: 'milestone', message: 'Phase 1 evidence submitted for review', time: '1 week ago', campaign_id: 'c1' },
];

File: backend/seed.py
"""
ClearFund Seed Script
Run with: python seed.py

Seeds campaigns and milestones only.
All SOL amounts start at 0 — real on-chain activity drives the numbers.
No fake donations, no fake transaction IDs.
Uses upsert logic — safe to run multiple times without duplicates.
"""

import asyncio
import os
from datetime import datetime, timezone, timedelta

from dotenv import load_dotenv
from motor.motor_asyncio import AsyncIOMotorClient

load_dotenv()

MONGO_URL = os.getenv("MONGO_URL", "mongodb://localhost:27017")
DB_NAME = os.getenv("DB_NAME", "clearfund")

# Real devnet vault address — fund with: solana airdrop 2 <address> --url devnet
VAULT_ADDRESS = "2hdLtoGgxfbxqgrVFqNiAy7G14LDyd4ePPHvV8Ti16Vo"

NOW = datetime.now(timezone.utc)


async def seed():
    client = AsyncIOMotorClient(MONGO_URL)
    db = client[DB_NAME]

    print(f"[Seed] Connected to {DB_NAME}")

    # ── NGO user (placeholder — gets replaced when real user logs in via Auth0)
    ngo = await db.users.find_one({"role": "ngo"})
    if not ngo:
        result = await db.users.insert_one({
            "email": "ngo@clearfund.app",
            "role": "ngo",
            "auth0_sub": "seed|ngo",
            "privy_wallet_id": None,
            "wallet_address": None,
            "created_at": NOW,
        })
        ngo_id = str(result.inserted_id)
        print("  [User] ngo@clearfund.app — inserted")
    else:
        ngo_id = str(ngo["_id"])
        print("  [User] ngo@clearfund.app — exists")

    # ── Campaigns ─────────────────────────────────────────────────────────────
    campaigns_data = [
        {
            "slug": "rural-health-van",
            "title": "Rural Health Van Project",
            "description": (
                "Deploying 3 mobile medical units to underserved rural communities "
                "in Maharashtra, India. Each van serves 5 villages per week, providing "
                "free consultations, diagnostics, and essential medicines to over "
                "15,000 people who have no access to healthcare facilities."
            ),
            "category": "Healthcare",
            "target_sol": 45.0,
            "vault_address": VAULT_ADDRESS,
            "ngo_name": "HealthForAll Foundation",
            "milestones": [
                {
                    "title": "Vehicle Purchase & Outfitting",
                    "description": (
                        "Purchase 3 second-hand ambulance-grade vans and outfit each "
                        "with a medical examination table, basic diagnostic equipment "
                        "(BP monitor, glucometer, pulse oximeter), a refrigeration unit "
                        "for vaccines, and first aid supplies."
                    ),
                    "amount_sol": 18.0,
                    "due_date": "2025-06-30",
                },
                {
                    "title": "Medical Staff Training",
                    "description": (
                        "Train 12 paramedics and 3 supervising doctors in mobile clinic "
                        "operations, rural triage protocols, and telemedicine. Includes "
                        "a 2-week intensive program run by Apollo Hospital volunteers."
                    ),
                    "amount_sol": 12.0,
                    "due_date": "2025-08-31",
                },
                {
                    "title": "First 90-Day Deployment",
                    "description": (
                        "Launch operations across 15 target villages. Track patient "
                        "visits, diagnoses, and referrals. Provide monthly public "
                        "impact reports with photo evidence submitted on-chain."
                    ),
                    "amount_sol": 15.0,
                    "due_date": "2025-11-30",
                },
            ],
        },
        {
            "slug": "solar-water-pumps",
            "title": "Solar Water Pumps Initiative",
            "description": (
                "Installing solar-powered water pumps in 20 drought-affected villages "
                "in Rajasthan. Each pump replaces 4-6 hours of daily manual water "
                "collection, freeing women and children to attend school and work. "
                "Covers installation, maintenance training, and 1-year warranty support."
            ),
            "category": "Water & Sanitation",
            "target_sol": 30.0,
            "vault_address": VAULT_ADDRESS,
            "ngo_name": "CleanWaterNow",
            "milestones": [
                {
                    "title": "Equipment Procurement & Import",
                    "description": (
                        "Source and import 20 high-efficiency solar pump kits from "
                        "certified manufacturers. Each kit includes solar panels, "
                        "submersible pump, controller, and 500L storage tank. "
                        "Delivery receipts and import docs submitted as evidence."
                    ),
                    "amount_sol": 15.0,
                    "due_date": "2025-07-31",
                },
                {
                    "title": "Installation & Community Training",
                    "description": (
                        "Install all 20 pumps with local contractor teams. Train 2 "
                        "community members per village in basic maintenance. Final "
                        "milestone evidence: geotagged installation photos and signed "
                        "community handover certificates."
                    ),
                    "amount_sol": 15.0,
                    "due_date": "2025-09-30",
                },
            ],
        },
        {
            "slug": "womens-shelter",
            "title": "Women's Shelter Renovation",
            "description": (
                "Renovating a 3,000 sq ft building in Chennai to serve as a full-time "
                "shelter for survivors of domestic violence. Will house 40 women and "
                "children with private rooms, a counselling centre, a skills training "
                "workshop, and a safe play area for children."
            ),
            "category": "Social Services",
            "target_sol": 25.0,
            "vault_address": VAULT_ADDRESS,
            "ngo_name": "SafeHaven Trust",
            "milestones": [
                {
                    "title": "Structural Renovation & Materials",
                    "description": (
                        "Complete all structural work: roof repair, waterproofing, "
                        "electrical rewiring, plumbing, and partition walls for private "
                        "rooms. Evidence: contractor invoices, before/after photos "
                        "with timestamps."
                    ),
                    "amount_sol": 10.0,
                    "due_date": "2025-06-30",
                },
                {
                    "title": "Interior Fit-Out & Furnishing",
                    "description": (
                        "Install beds, mattresses, wardrobes, kitchen appliances, "
                        "counselling room furniture, and children's play equipment. "
                        "Evidence: purchase receipts and installation photos."
                    ),
                    "amount_sol": 10.0,
                    "due_date": "2025-08-31",
                },
                {
                    "title": "Shelter Launch & First Month Operations",
                    "description": (
                        "Open shelter to residents. Provide evidence of first cohort "
                        "admitted (anonymised), staff hired, and first month operations "
                        "report signed by a certified social worker."
                    ),
                    "amount_sol": 5.0,
                    "due_date": "2025-10-31",
                },
            ],
        },
        {
            "slug": "youth-stem-lab",
            "title": "Youth STEM Lab",
            "description": (
                "Equipping a fully functional STEM lab for 200 underprivileged youth "
                "aged 12-18 in Bengaluru. The lab will run weekend coding, robotics, "
                "and electronics workshops taught by volunteer engineers from local "
                "tech companies. Goal: 50 students placed in tech internships by year 2."
            ),
            "category": "Education",
            "target_sol": 20.0,
            "vault_address": VAULT_ADDRESS,
            "ngo_name": "EduFuture",
            "milestones": [
                {
                    "title": "Lab Equipment & Setup",
                    "description": (
                        "Purchase 20 laptops, 10 Arduino/Raspberry Pi robotics kits, "
                        "a 3D printer, networking equipment, and lab furniture. "
                        "Evidence: purchase receipts, delivery photos, and lab "
                        "readiness sign-off from EduFuture director."
                    ),
                    "amount_sol": 12.0,
                    "due_date": "2025-07-31",
                },
                {
                    "title": "Launch Cohort & First 3-Month Program",
                    "description": (
                        "Enrol first cohort of 50 students. Run 12-week program with "
                        "weekly sessions. Evidence: attendance registers, student "
                        "project photos, and end-of-cohort demo day video."
                    ),
                    "amount_sol": 8.0,
                    "due_date": "2025-10-31",
                },
            ],
        },
    ]

    for c in campaigns_data:
        slug = c.pop("slug")
        milestones_def = c.pop("milestones")

        campaign_doc = {
            "ngo_id": ngo_id,
            "title": c["title"],
            "description": c["description"],
            "category": c["category"],
            "target_sol": c["target_sol"],
            "total_raised_sol": 0.0,
            "vault_address": c["vault_address"],
            "ngo_name": c["ngo_name"],
            "status": "active",
            "trust_score": 0.0,
            "failure_count": 0,
            "created_at": NOW - timedelta(days=7),
        }

        result = await db.campaigns.update_one(
            {"title": c["title"]},
            {"$setOnInsert": campaign_doc},
            upsert=True,
        )
        campaign_doc_db = await db.campaigns.find_one({"title": c["title"]})
        campaign_id = str(campaign_doc_db["_id"])
        action = "inserted" if result.upserted_id else "exists"
        print(f"  [Campaign] {c['title']} — {action}")

        for m in milestones_def:
            milestone_doc = {
                "campaign_id": campaign_id,
                "title": m["title"],
                "description": m["description"],
                "amount_sol": m["amount_sol"],
                "due_date": m["due_date"],
                "status": "pending",
                "evidence_urls": [],
                "ai_decision": {},
                "oracle_result": {},
                "solana_tx": None,
                "created_at": NOW - timedelta(days=6),
            }
            result = await db.milestones.update_one(
                {"campaign_id": campaign_id, "title": m["title"]},
                {"$setOnInsert": milestone_doc},
                upsert=True,
            )
            action = "inserted" if result.upserted_id else "exists"
            print(f"    [Milestone] {m['title']} — {action}")

    print("\n[Seed] Done. All SOL balances start at 0 — fund the vault with real devnet SOL.")
    print(f"[Seed] Vault address: {VAULT_ADDRESS}")
    print("[Seed] Run: solana airdrop 2 <vault_address> --url devnet")
    client.close()


if __name__ == "__main__":
    asyncio.run(seed())
